src/Http/Http.Extensions/src/ProblemDetailsService.cs

```diff
@@ -15,44 +15,33 @@ internal sealed class ProblemDetailsService : IProblemDetailsService
         _writers = writers.ToArray();
     }
 
-    public ValueTask WriteAsync(ProblemDetailsContext context)
+    public async ValueTask WriteAsync(ProblemDetailsContext context)
     {
-        ArgumentNullException.ThrowIfNull(context);
-        ArgumentNullException.ThrowIfNull(context.ProblemDetails);
-        ArgumentNullException.ThrowIfNull(context.HttpContext);
-
-        if (context.HttpContext.Response.HasStarted ||
-            context.HttpContext.Response.StatusCode < 400 ||
-            _writers.Length == 0)
+        if (!await TryWriteAsync(context))
         {
-            return ValueTask.CompletedTask;
+            throw new InvalidOperationException("Unable to find a registered `IProblemDetailsWriter` that can write to the given context.");
```

Comment:
Throwing is a breaking behavior change?

Reply:
It is but I believe it is consistent with TryParse/Parse approach. However, I ok to not do that and avoid this breaking change